storage_service/locations/models/package.py

```diff
@@ -559,6 +560,130 @@ def compress_package(self, algorithm, extract_path=None):
 
         return (compressed_filename, extract_path)
 
+    def _parse_mets(self, prefix=None, relative_path=['metadata', 'submissionDocumentation', 'METS.xml']):
+        """
+        Parses a transfer's METS file, and returns a dict with metadata about
+        the transfer and each file it contains.
+
+        :param prefix: The location of the transfer containing the METS file
+            to parse. If not provided, self.full_path is used.
+        :param relative_path: An array containing one or more path components.
+            These will be joined together with the prefix to produce the
+            complete path of the METS within this transfer.
+        :return: A dict in the following structure:
+            {
+                "transfer_uuid": "The UUID of the originating transfer",
+                "creation_date": "The date the transfer METS was created, in ISO 8601 format",
+                "dashboard_uuid": "The UUID of the originating dashboard",
+                "files": [
+                    # An array of zero or more dicts representing file
+                    # metadata, in the following format:
+                    {
+                        "file_uuid": "The UUID of the file",
+                        "path": "The full path of the file within the transfer,
+                                 including the transfer's root directory."
+                    }
+                ]
+            }
+        :raises StorageException: if the requested METS file cannot be found,
+            or if required elements are missing.
+        """
+        if prefix is None:
+            prefix = self.full_path
+
+        mets_path = os.path.join(prefix, *relative_path)
+        if not os.path.exists(mets_path):
+            raise StorageException("No METS found at location: {}".format(mets_path))
+
+        doc = etree.parse(mets_path)
+
+        mets_ns = utils.NSMAP['mets']
+        namespaces = {'m': mets_ns}
+        mets = doc.xpath('/m:mets', namespaces=namespaces)
```

Comment:
`doc.find` doesn't implement the full XPATH syntax (while `doc.xpath`) does, so not all of these queries work with `.find`. I'll change the ones that do though.